src/smartbuffer.ts

```diff
@@ -749,6 +749,16 @@ class SmartBuffer {
         this.clear();
     }
 
+    /**
+     * Ensures that the internal Buffer is large enough to read data.
+     *
+     * @param length { Number } The length of the data that needs to be read.
+     */
+    ensureReadable(length) {
```

Comment:
From a style standpoint, could you please refactor this to be a private method and add the number type to length.

```typescript
private ensureReadable(length: number) {
```